src/ORM/Association/HasMany.php

```diff
@@ -117,6 +165,10 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
                 $targetEntity->set($properties, ['guard' => false]);
             }
 
+            if ($this->_saveStrategy === self::SAVE_REPLACE) {
+                $targetEntity->isNew(true);
```

Comment:
Yes, there is. I have an idea. I will apply the delete action only for the ones that have primary keys different from the ones that are remaining in the list.
I will code that.